### PR TITLE
Fix getTodayWord bug

### PR DESCRIPTION
**Pull request by ganneo**

### CODE REVIEW

**Review by tiffanygan — APPROVED** [review]
lgtm

File: src/service/WordService.js [hunk 1]
```diff
@@ -44,11 +44,8 @@ export default class WordService {
         return true;
       }
 
-      const lastProgressDate = new Date(lastProgress.time);
-      const todayDate = Util.getTodayDate();
-      const dayDifference = Math.floor((todayDate - lastProgressDate) / SECONDS_IN_A_DAY);
-
-      return MEMORIZATION_DATES.includes(dayDifference);
+      const daysDone = progresses.filter((progress) => progress.pass === true).length
+      return MEMORIZATION_DATES.includes(daysDone);
```
Comment (yugan2005) [inline src/service/WordService.js]: `daysDone` only gives the index for getting the current value from `MEMORIZATION_DATES`
You still need check whether the `dayDifference` is equal to that value or not

**Review by yugan2005 — CHANGES_REQUESTED** [review]

File: src/service/WordService.js [hunk 2]
```diff
@@ -44,11 +44,8 @@ export default class WordService {
         return true;
       }
 
-      const lastProgressDate = new Date(lastProgress.time);
-      const todayDate = Util.getTodayDate();
-      const dayDifference = Math.floor((todayDate - lastProgressDate) / SECONDS_IN_A_DAY);
-
-      return MEMORIZATION_DATES.includes(dayDifference);
+      const daysDone = progresses.filter((progress) => progress.pass === true).length
```
Comment (tiffanygan) [inline src/service/WordService.js]: `progress.pass === true` is the same as `progress.pass`

**Review by tiffanygan — CHANGES_REQUESTED** [review]

**Review by yugan2005 — APPROVED** [review]

File: src/service/WordService.js [hunk 3]
```diff
@@ -34,24 +34,26 @@ export default class WordService {
...
+      const daysDone = progresses.filter((progress) => progress.pass).length;
+      const requiredMemorizationDaysApart = MEMORIZATION_DATES[daysDone - 1];
+      const actualDaysApart = (Util.getTodayDate() - word.createTime) / SECONDS_IN_A_DAY;
 
-      return MEMORIZATION_DATES.includes(dayDifference);
+      return actualDaysApart >= requiredMemorizationDaysApart;
     });
 
-    return todayWordArray;
+    return todayWordArray.sort(() => Math.random - 0.5);
```
Comment (tiffanygan) [inline src/service/WordService.js]: `Math.random` needs a bracket after it. `Math.random()`

**Review by tiffanygan — CHANGES_REQUESTED** [review]

**Review by tiffanygan — APPROVED** [review]

**Pull request merged**